### PR TITLE
chore: Change `generate_library_integration_test.sh `to test on multiple libraries in `googleapis-gen`

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Change `generate_library_integration_test.sh` to test on multiple libraries in `googleapis-gen`. The libraries are chosen from [here](https://github.com/googleapis/sdk-platform-java/tree/main/test/integration/goldens).

### CODE REVIEW

File: library_generation/test/resources/proto_path_list.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# This file is used in integration test against `generate_library.sh`.
+# Format:
+# proto_path destination_path
+google/bigtable/v2 google-cloud-bigtable-v2-java
+google/cloud/apigeeconnect/v1 google-cloud-apigeeconnect-v1-java
+google/cloud/asset/v1 google-cloud-asset-v1-java
+google/cloud/compute/v1 google-cloud-compute-v1-java
+google/cloud/kms/v1 google-cloud-kms-v1-java
+google/example/library/v1 google-cloud-example-library-v1-java
+# google/iam/v1 google-iam-v1-java
```
Comment (JoeWang1127) [inline library_generation/test/resources/proto_path_list.txt:10]: Temporarily exclude to generate `iam` due to #1992, will address the issue in another PR.

File: library_generation/test/generate_library_integration_test.sh [hunk 2]
```diff
@@ -56,42 +58,49 @@ protobuf_version=$(get_version_from_WORKSPACE "protobuf-" WORKSPACE "-")
...
+  # include gapic_metadata.json and package-info.java after
+  # resolving https://github.com/googleapis/sdk-platform-java/issues/1986
+  diff -r "googleapis-gen/${proto_path}/${destination_path}" "${output_folder}/${destination_path}" -x "*gradle*" -x "gapic_metadata.json" -x "package-info.java" || RESULT=$?
 
-popd # output_folder
+  if [ ${RESULT} == 0 ] ; then
+    echo "SUCCESS: Comparison finished, no difference is found."
+  else
+    echo "FAILURE: Differences found in proto path: ${proto_path}."
+    exit "${RESULT}"
```
Comment (diegomarquezp) [inline library_generation/test/generate_library_integration_test.sh:101]: Would it make sense to build a list of failed libraries and then exit if the list is non emptyto have better insight on potential problems?

Comment (JoeWang1127) [reply library_generation/test/generate_library_integration_test.sh:101]: I think the integration tests here is to mimic what happens in the real world, so we need to modify `generate_library.sh` if any difference exists.

We do have unit tests to verify failed cases (e.g., invalid versions).

**Review by blakeli0 — APPROVED** [review]
LGTM. Good to see that we found one more edge case. Please still try to test the script against all libraries at least once as we discussed in the project status meeting.

**Review by meltsufin — APPROVED** [review]
